### DISCUSSION TITLE
Image renders differently on userpage, repository and metrics page

### DISCUSSION THREAD

**Discussion by oSumAtrIX** [lowlighter/metrics#803] [Q&A] [answered]
I expect the metrics to render like [this](https://github.com/oSumAtrIX/oSumAtrIX), but they render like [that](https://github.com/oSumAtrIX).
Re: #374, #787

Additionally, the prefered license is different on the metrics page from what is displayed on Github.

**Comment by lowlighter** [comment]
Hi 👋 !

> I expect the metrics to render like this, but they render like that.

Both link seems to yield the same render for me so I'm not sure what is highlighted 😕 

But seems you use `config_displays: columns` which is indeed broken as #374 suggests and the only workaround for now is to split it manually in 2 columns and use `<img align="left">` and `<img align="right">` 

> Additionally, the prefered license is different on the metrics page from what is displayed on Github.

I'm not aware of this stats displayed directly on GitHub, could you provide a screenshot of what you meant ?
The one displayed is an overall stat for all your repositories (which also depends on how `repositories_affiliations` was configured), not just a single one

**Comment by oSumAtrIX** [comment, marked as answer]
Hey,

Thanks for checking back. 
> Both link seems to yield the same render for me so I'm not sure what is highlighted 😕

You are right. Seems like it was an issue on my end for a short period.
> I'm not aware of this stats displayed directly on GitHub

Metrics allows to include the prefered license which a user uses. If you use my name on GitHub, it will display MIT. Though, my repositories are mostly GPLv3, which is displayed correctly on [the metrics](https://metrics.lecoq.io), but not correctly, when using the workflow: 
![image](https://user-images.githubusercontent.com/13122796/151031221-2e1ceba4-185a-4f41-b90b-9c372e3cf8d4.png)

I noticed similar issues of displaying the prefered indentation style, which for me is tabs. Unfortunately, I can not use the site right now due to this issue, which was happening couple days ago as well:
![image](https://user-images.githubusercontent.com/13122796/151031812-7347478c-702d-4e01-b671-4a860309e71c.png)

**Reply by lowlighter** [reply]
> Metrics allows to include the prefered license which a user uses. If you use my name on GitHub, it will display MIT. Though, my repositories are mostly GPLv3, which is displayed correctly on the metrics, but not correctly, when using the workflow:

I see.
I think it's caused because in your workflow you use `repositories_forks: yes`, which count forked repositories in your stats and it seems that most of them are MIT ultimately shifting the balance:

![image](https://user-images.githubusercontent.com/22963968/151035789-d669e1f1-5cd2-45ef-952b-910dda7f2f9f.png) ![image](https://user-images.githubusercontent.com/22963968/151036218-229d6497-70f3-41db-b840-7b5a40de5444.png)

(sorry for color change 😅)

> I noticed similar issues of displaying the prefered indentation style, which for me is tabs

IIRC it fetches your last push events, match `\s+` and count whethers it's spaces or tabs. It doesn't actually look in your repositories so it may varies over time, especially if you contribute to other projects that don't use the same indent style as you. It's not an exact science 😅 

Additionally some slight differences can appear between the website and the action, depending on the scopes you granted to your token.
The website don't have access to any private data/repositories/events/etc. and is limited to only what a scopeless token can fetch about other users

> Unfortunately, I can not use the site right now due to this issue, which was happening couple days ago as well

Yes it was reported in #805, I'll investigate later today or this week

> I expect the metrics to render like this, but they render like that.

If you need a workaround for this, you can check my [`readme.md`](https://github.com/lowlighter/lowlighter/blob/7fc32cf5483c9bc87ef419030533266d1f3845ab/README.md?plain=1#L1-L7) (I use the `<img align>` trick) to give you an idea.
Within it there's also a link towards an placeholder gist svg that you can size however you want to "fill gaps" and make the alignement correct.

It's not perfect but it does the trick

___

Thanks again for using this 🙂 !
